.github/CONTRIBUTING.md

```diff
@@ -1,2 +1,63 @@
+
+## Code contributions
+Much of the WRF-Hydro development occurs on the public [NCAR/wrf_hydro_nwm_public](https://github.com/NCAR/wrf_hydro_nwm_public) repository. However, some internal development must occur on a private repository maintained by NCAR and NOAA. Due to this dual-repository system, a strict procedure must be followed when making contributions to the WRF-Hydro model code.
+
+All code development contributions will be made via [forks](https://help.github.com/articles/about-forks/) and [Pull Requests](https://help.github.com/articles/about-pull-requests/). If you are unfamiliar with GitHub, forks, or pull requests, see [collaborating with issues and pull requests](https://help.github.com/categories/collaborating-with-issues-and-pull-requests/) for a thorough guide to collaborating on GitHub.
+
+### Fortran code style standards
+Please see our Fortran [code style guidelines](CODESTYLE.md)
+
+### Universal guidelines
+
+* All code contributions must be made through GitHub following specific guidelines listed below.
+* All contributions must be made in an open, and public way via the GitHub repository. There will be no embargo period for code contributions.
+* All contributions must be able to be merged without conflict. It is the responsibility of the contributor to resolve any merge conflicts **prior** to submitting a Pull Request.
+* All contributions must pass relevant automated testing procedures. These tests are executed automatically when you submit your pull request.
+* Be respectful when giving and receiving feedback on contributions or issues. 
+
+### Pull requests
+
+#### Contributor responsibility
+Pull requests may be denied for a variety of reasons. It is the responsibility of contributors making pull requests to follow the best practices. Not following best practices described below is grounds for rejection of pull requests and may result in excessive manual work for contributors. It is the contributors’ responsibility to know and follow best git practices.
+
+#### Commits: practices & messages
+Learning to use commits wisely and “atomically” is the foundation for success. As users gain familiarity with basics of how git works and the merging practices we are using. The next level of sophistication is interaction with the git log. Reading logs is greatly facilitated by writing good logs, or “commit messages”. The following webpage gives a crash course on how to write good commit messages: https://chris.beams.io/posts/git-commit/
+
+#### Frequency
+Pull requests should happen with high enough frequency to minimize the scope of the code review. Commits which do not changes answers (pass regression testing) MUST be kept separate from commits which do change answers even when the work is on the same feature or bug. The portion of the code which changes the answer must be isolated as much as possible into its own pull request.
```

Comment:
This is a very noble goal. It may be worth illustrating that this means code should be refactored and readied to modify behavior prior to or after the behavior change. It may also be worth noting that this facilitates review and trust of behavior changes.